routes/artifacts.js

```diff
@@ -386,18 +387,18 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
         if (this.artifactRegion === 'us-east-1') {
           vhost = 's3.amazonaws.com';
         } else {
-          vhost = `s3-${this.canonicalArtifactRegion}.amazonaws.com`;
+          vhost = `s3-${this.artifactRegion}.amazonaws.com`;
         }
 
         // Generate the public URL for this artifact
         var canonicalArtifactUrl = urllib.format({
           protocol: 'https:',
-          host: vhost,
+          host: bucket + '.' + vhost,
           pathname: prefix,
         });
         console.log('CLOUDMIRROR canonical artifact url: ' + canonicalArtifactUrl);
 
-        if (this.artifactRegion === region) {
+        if (!region || this.artifactRegion === region) {
           // Since we don't need a redirect, we just set things to what they
           // are in the canonical region
           url = canonicalArtifactUrl;
```

Comment:
Still wrong...

We don't get CDN, we need to use `this.publicBucket.createGetUrl(prefix);`